.github/workflows/carpenter.yaml

```diff
@@ -28,7 +45,7 @@ jobs:
   carpenter-build-dev:
     name: carpenter_build_dev
     runs-on: ubuntu-latest
-    if: github.ref != 'refs/heads/main'
```

Comment:
So we'll only "build_dev" off of the "head" of `main`, and when the GitHub event name is not "release", but line 23 will release off of any `github.ref` if the `github.event_name` is release and push that release to `main` (see line 31).

Can we put `refs/heads/<name>` in a variable instead, and then we can "release" off of any branch?

Reply:
I left the branch, tag, and release test artifacts in the [sysbench repo](https://github.com/arcalot/arcaflow-plugin-sysbench), and the test images in [quay](https://quay.io/repository/arcalot/arcaflow-plugin-sysbench?tab=tags), for now in case you want to see them. I'll clean those up after this is resolved.

Reply:
You can also check the sysbench [carpenter action history](https://github.com/arcalot/arcaflow-plugin-sysbench/actions/workflows/carpenter.yaml).

Reply:
I agree that the repeats are sub-optimal. I haven't checked deeply into the logic options that are available for flattening it. I'll check it out with your example.

Reply:
I tested the collapsed approach, and the downside is that `steps` are not reflected individually in the Github actions tracking the way that `jobs` are. So the collapsed approach simply shows a carpenter build job is happening without any context to validate which build between `release`, `prod`, and `dev`. IMO that lost context isn't worth the deduplication of code.

There may be another approach where we could run individual jobs to set the env vars, and one final global job to do the build, but even though this reduces duplication it honestly just creates more code that's harder to follow. Jobs are in independent environments, so we have to insert extra `outputs` declarations for the jobs to pass the vars, and then we need a conditional in the build job, along with logic to ensure the build only runs after the jobs that set the vars. It's ugly.

I think we should stick with the proposed PR here as designed.

Reply:
Can we put `${{ github.event_name }}` in with the job name (see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#run-name), e.g.:
```
  carpenter-build:
    name: carpenter build
    run-name: Build image for ${{ github.event_name }} from ${{ github.ref }}
    runs-on: ubuntu-latest
    steps:
      - name: set image tag for releases
        if: github.event_name == 'release'
        run: |
          echo "IMAGE_TAG=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
      - name: set image tag for dev builds
        if: github.event_name != 'release' && github.ref != 'refs/heads/main'
        run: |
          export commit_hash=${{ github.sha }}
          echo "IMAGE_TAG=${GITHUB_REF##*/}_${commit_hash:0:7}" >> $GITHUB_ENV
      - name: set quay image expiration for dev builds
        if: github.event_name != 'release' && github.ref != 'refs/heads/main'
        run: |
          echo "QUAY_IMG_EXP=90d" >> $GITHUB_ENV
      - name: Checkout this project
        uses: actions/checkout@v3
      - name: carpenter build step
        uses: arcalot/arcaflow-plugin-image-builder@main
        with:
          args: build --build --push
```

Reply:
I've come up with and tested a compromise that I can live with. Please have another look.